rllib/agents/trainer.py

```diff
@@ -914,19 +952,32 @@ def step_attempt(self) -> ResultDict:
                 with concurrent.futures.ThreadPoolExecutor() as executor:
                     train_future = executor.submit(
                         lambda: next(self.train_exec_impl))
-                    if self.config["evaluation_num_episodes"] == "auto":
-
-                        # Run at least one `evaluate()` (num_episodes_done
-                        # must be > 0), even if the training is very fast.
-                        def episodes_left_fn(num_episodes_done):
-                            if num_episodes_done > 0 and \
+                    # Automatically determine duration of the evaluation.
+                    if self.config["evaluation_duration"] == "auto":
+                        unit = self.config["evaluation_duration_unit"]
+
+                        # Run at least one `evaluate()`, even if the training
+                        # is very fast.
+                        def duration_fn(remaining_duration):
```

Comment:
should we define this duration_fn outside of all these if clauses, higher up in this function, so it doesn't have to live in such a nested place?
you can partial bind the train_future instance for example.